rasa/nlu/tokenizers/tokenizer.py

```diff
@@ -0,0 +1,45 @@
+from typing import Text, List, Optional, Dict, Any
+
+from rasa.nlu.components import Component
+from rasa.nlu.constants import (
+    MESSAGE_RESPONSE_ATTRIBUTE,
+    MESSAGE_TEXT_ATTRIBUTE,
+    CLS_TOKEN,
+)
+
+
+class Token(object):
+    def __init__(self, text, offset, data=None):
+        self.offset = offset
+        self.text = text
+        self.end = offset + len(text)
+        self.data = data if data else {}
+
+    def set(self, prop, info):
+        self.data[prop] = info
+
+    def get(self, prop, default=None):
+        return self.data.get(prop, default)
+
+
+class Tokenizer(Component):
+    def __init__(self, component_config: Optional[Dict[Text, Any]] = None) -> None:
+        super(Tokenizer, self).__init__(component_config)
+
+        if "use_cls_token" in self.component_config:
+            self.use_cls_token = self.component_config["use_cls_token"]
+        else:
+            self.use_cls_token = False
```

Comment:
let's add `use_cls_token` to all defaults though, so that all defaults config is clearly visible when you open the class file.

I think it makes sense to set it to `True` by default because in the new architecture, if it is not present, last word will be used for intent classification, which can create unnecessary bias towards it. I can imagine that we want to remove it if we want to use old style intent classifier, that's why let's keep it configurable